safe/utilities/direction_distance.py

```diff
@@ -0,0 +1,84 @@
+# coding=utf-8
+
+"""Direction and Distance Functionality."""
+
+import logging
+from qgis.core import QgsDistanceArea
+
+LOGGER = logging.getLogger('InaSAFE')
+
+
+def bearing_to_cardinal(angle):
+    """Get cardinality of an angle.
+
+    :param angle: Bearing angle
+    :return:
+    """
+    direction_list = [
+        'N', 'NNE', 'NE', 'ENE', 'E', 'ESE', 'SE',
+        'SSE', 'S', 'SSW', 'SW', 'WSW', 'W',
+        'WNW', 'NW', 'NNW'
+    ]
+
+    bearing = float(angle)
+    direction_count = len(direction_list)
+    direction_interval = 360. / direction_count
+    index = int(round(bearing / direction_interval))
+    index %= direction_count
+    return direction_list[index]
+
+
+def distance_direction(hazard_point, places_layer, name, population=False):
+    """Calculate distance and bearing angle.
+
+    :param hazard_point:
+    :param places_layer:
+    :param name:
+    :param population:
+    :return:
+    """
+    # define distance
+    find_distance = QgsDistanceArea()
+    find_distance.setEllipsoidalMode(True)
+    find_distance.setEllipsoid('WGS84')
+
+    fields = places_layer.dataProvider().fields()
+    # list to store cities information
+    cities = []
+    # start calculating distance and get cardinality
+    for feature in places_layer.getFeatures():
+        city_name = str(feature[fields.indexFromName(name)])
```

Comment:
I would avoid the `str` function. What about if the city is called Besançon, the city where I was born ;-)
You will get a python exception ;-)

Reply:
Hmmm, I am not aware if that will give me an exception. Thanks for the information.